extensions/amp-list/0.1/amp-list.js

```diff
@@ -345,46 +352,20 @@ export class AmpList extends AMP.BaseElement {
   }
 
   /**
-   * Returns true if element's src points to amp-state.
-   *
-   * @param {string} src
-   * @return {boolean}
-   * @private
-   */
-  isAmpStateSrc_(src) {
-    return (
-      isExperimentOn(this.win, 'amp-list-init-from-state') &&
-      startsWith(src, AMP_STATE_URI_SCHEME)
-    );
-  }
-
-  /**
-   * Gets the json an amp-list that has an "amp-state:" uri. For example,
-   * src="amp-state:json.path".
+   * Gets optionally provided initialization json to use for initial render.
    *
-   * @param {string} src
-   * @return {Promise<!JsonObject>}
+   * @return {null|Promise<?JsonObject>}
    * @private
    */
-  getAmpStateJson_(src) {
-    return Services.bindForDocOrNull(this.element)
-      .then(bind => {
-        userAssert(bind, '"amp-state:" URLs require amp-bind to be installed.');
-        userAssert(
-          !this.ssrTemplateHelper_.isEnabled(),
-          '[amp-list]: "amp-state" URIs cannot be used in SSR mode.'
-        );
-
-        const ampStatePath = src.slice(AMP_STATE_URI_SCHEME.length);
-        return bind.getState(ampStatePath);
-      })
-      .then(json => {
-        userAssert(
-          typeof json !== 'undefined',
-          `[amp-list] No data was found at provided uri: ${src}`
-        );
-        return json;
-      });
+  getInitialJson_() {
+    if (!isExperimentOn(this.win, 'amp-list-init-from-state')) {
+      return null;
+    }
+    if (this.initialJsonUsed_ || !hasChildJsonConfig(this.element)) {
+      return null;
+    }
+    this.initialJsonUsed_ = true;
+    return Promise.resolve(getChildJsonConfig(this.element));
```

Comment:
`getChildJsonConfig` can throws some uncaught errors. I'd recommend reducing these to warnings since the fetched data is still available:
```js
try {
  return Promise.resolve(getChildJsonConfig(this.element));
} catch (e) {
  user().warn(
    TAG,
    'Failed to initialize with the following message: [%s]. Rendering from "src" instead.',
    e.message
  );
  return null;
}
```